### PR TITLE
Add workflow for building GEOS and populate relevant tasks

### PR DESCRIPTION
**Pull request by danholdaway**
## Description

Add tasks for clone, build (by linking) and build GEOS.

Add workflow for building GEOS. First part of having a workflow for making a forecast with GEOS.

### CODE REVIEW

**Comment by danholdaway** [comment]
@mathomp4 would appreciate if you have a chance to check over the `clone_geos.py ` and `build_geos.py` tasks to make sure I haven't done something incorrect.

File: src/swell/utilities/build.py [hunk 1]
```diff
@@ -16,22 +16,22 @@
...
-    jedi_bundle_source_path = os.path.join(jedi_bundle_path, 'source')
+    package_build_path = os.path.join(package_path, 'build')
+    package_source_path = os.path.join(package_path, 'source')
 
     # Remove trailing slash if needed
-    if jedi_bundle_build_path.endswith('/'):
-        jedi_bundle_build_path = jedi_bundle_build_path[:-1]
-    if jedi_bundle_source_path.endswith('/'):
-        jedi_bundle_source_path = jedi_bundle_source_path[:-1]
+    if package_build_path.endswith('/'):
```
Comment (jardizzo) [inline src/swell/utilities/build.py]: You might be able to just use rstrip('/') here.

**Review by jardizzo — APPROVED** [review]
Looks like lots of subprocess code consolidation in addition to new tasks. I need to make sure that the nightly CI workflow(s) are not coupled with a now outdated modules file since a new module load was added for GEOS. See one minor comment on possible benefit of "rstrip()" to remove trailing characters (can eliminate the endswith check).

**Comment by Dooruk** [comment]
The GEOS build worked for me. Only issue (not related to this PR) I had was the long wait time for the compute node.

**Review by Dooruk — APPROVED** [review]
The GEOS build worked for me. Only issue (not related to this PR) I had was the long wait time for the compute node.

File: src/swell/utilities/build.py (hunk 1 above)
Comment (danholdaway) [reply src/swell/utilities/build.py]: Thanks @jardizzo, always happy to eliminate an `if`!

**Pull request merged**